sql/table.go

```diff
@@ -393,11 +392,10 @@ func makeKeyVals(desc *TableDescriptor, columnIDs []ColumnID) ([]parser.Datum, e
 }
 
 func decodeIndexKeyPrefix(desc *TableDescriptor, key []byte) (IndexID, []byte, error) {
-	if !bytes.HasPrefix(key, keys.TableDataPrefix) {
+	if encoding.PeekType(key) != encoding.Int {
 		return 0, nil, util.Errorf("%s: invalid key prefix: %q", desc.Name, key)
```

Comment:
You'll still be able to see what the key prefix is and that it isn't a valid int tag. Do you have a suggestion for something better?